src/main/java/io/vlingo/schemata/model/Context.java

```diff
@@ -29,14 +30,14 @@ static Context with(
           final ContextId contextId,
           final String name,
           final String description) {
-    final Context context = stage.actorFor(Context.class, ContextEntity.class, contextId);
-    context.defineWith(name, name);
+    final Context context = stage.actorFor(Context.class, ContextEntity.class);
```

Comment:
The problem you are experiencing is here. All persistent actor instances must be instantiated with their unique identity. This is because the first message every `Actor` receives is `start()` and every entity base class attempts to load the persistent state of the actor during `start()`. If the identity is not there already then `start()` attempts to load from a null object. I am not sure what happens exactly, but there is likely a `NullPointerException` thrown in the entity or in the persistent `Store` being used.

See: https://github.com/vlingo/vlingo-lattice/blob/04cd492128b52af5c51b404432998cbbefd21be1/src/main/java/io/vlingo/lattice/model/object/ObjectEntity.java#L65

Reply:
Thanks, definitely it was that. I've fixed it. However one must know that it should override `whenNewState`, shouldn't be better this was ensured via constructor ?